generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx

```diff
@@ -20,31 +20,82 @@ import * as React from 'react';
 import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { Button } from 'reactstrap';
+import { Pagination } from 'react-bootstrap';
 import { Translate, ICrudGetAction, TextFormat } from 'react-jhipster';
-import { FaPlus, FaEye, FaPencil, FaTrash } from 'react-icons/lib/fa';
+import { FaPlus, FaEye, FaPencil, FaSort, FaTrash } from 'react-icons/lib/fa';
 
 import { getUsers } from '../../../reducers/user-management';
 import { APP_DATE_FORMAT } from '../../../config/constants';
+import { ITEMS_PER_PAGE } from '../../../shared/util/pagination.constants';
+import { getUrlParameter, IPaginationState } from '../../../shared/util/pagination-utils';
 
 export interface IUserManagementProps {
   getUsers: ICrudGetAction;
   users: any[];
   account: any;
   match: any;
+  totalItems: 0;
 }
 
-export class UserManagement extends React.Component<IUserManagementProps, undefined> {
+export class UserManagement extends React.Component<IUserManagementProps, IPaginationState> {
 
   constructor(props) {
     super(props);
+    const pageParam = getUrlParameter('page', props.location.search);
+    const sortParam = getUrlParameter('sort', props.location.search);
+    let sort = 'id';
+    let order = 'asc';
+    let activePage = 1;
+    if (pageParam !== '' && !isNaN(Number(pageParam))) {
+      activePage = Number(pageParam);
+    }
+    if (sortParam !== '') {
+      sort = sortParam.split(',')[0];
+      order = sortParam.split(',')[1];
+    }
+    this.state = {
+      itemsPerPage: ITEMS_PER_PAGE,
+      sort,
+      order,
+      activePage
+    };
   }
 
   componentDidMount() {
-    this.props.getUsers();
+    this.props.getUsers(this.state.activePage - 1, this.state.itemsPerPage, this.state.sort + ',' + this.state.order);
+  }
+
+  sort = prop => e => {
+    this.state.order === 'asc' ?
+      this.setState({
+        order: 'desc',
+        sort: prop
+    }, () => {
+      this.sortUsers();
+    }) :
+    this.setState({
+       order: 'asc',
+       sort: prop
+    }, () => {
+      this.sortUsers();
+    });
+  }
+
+  sortUsers() {
+    location.href = location.href.split('?')[0] + '?page=' + this.state.activePage + '&sort=' + this.state.sort + ',' + this.state.order;
```

Comment:
I guess its not a good idea to mutate location directly, we should use the api provided by React router to update this so that states are in sync

Reply:
can we use string literals instead of concatenation?